doc/pfc_asym/PFC_Asymmetric_Test_HLD.md

```diff
@@ -0,0 +1,423 @@
+# Asymmetric PFC Test Plan
+
+* [Overview](#Overview)
+   * [Scope](#Scope)
+   * [Testbed](#Testbed)
+* [Setup configuration](#setup-configuration)
+* [Existed modules refactoring](#existed-modules-refactoring)
+* [Python  modules to setup and run test](#python-modules-to-setup-and-run-test)
+   * [Python modules](#python-modules)
+   * [PFC storm global variables](#pfc-storm-global-variables)
+   * [PTF test case execution](#ptf-test-case-execution)
+   * [Pytest fixtures](#pytest-fixtures)
+		* [deploy_pfc_gen](#deploy_pfc_gen)
+		* [setup](#setup)
+		* [pfc_storm_template](#pfc_storm_template)
+		* [pfc_storm_runner](#pfc_storm_runner)
+		* [enable_pfc_asym](#enable_pfc_asym)
+* [Test](#Test)
+* [Test cases](#Test-cases)
+
+## Overview
+The purpose is to test functionality of Asymmetric PFC on the SONIC based DUT, closely resembling production environment.
+
+### Scope
+The test is targeting a running SONIC system with fully functioning configuration. The purpose of the test is to perform functional testing of Asymmetric PFC on SONIC system. There will be reused existed PTF test suite for PFC Asymmetric which is located at https://github.com/Azure/sonic-mgmt/blob/master/ansible/roles/test/files/saitests/pfc_asym.py.
+
+### Testbed
+The test will run on the following testbeds:
+* T0
+
+### DUT image
+RPC image is required
+
+## Setup configuration
+No setup pre-configuration is required, test will configure and clean-up all the configuration.
+
+## Existed modules refactoring
+There is already existed PFC asymmetric test cases which use PTF to send traffic.
+
+Test suite location
+
+https://github.com/Azure/sonic-mgmt/blob/master/ansible/roles/test/files/saitests/pfc_asym.py
+
+It requires several updates:
+
+1. Packets sending speed can be increased by using multiprocessing instead of multithreading library.
+
+https://github.com/Azure/sonic-mgmt/blob/master/ansible/roles/test/files/saitests/pfc_asym.py#L83
```

Comment:
can you explain why this is needed? what does it wishes to fix/handle?

Reply:
Theoretically I/O operations will be executed faster by using separate process rather then python threading because of python GIL.

Reply:
As test bed is very limited by traffic generation possibilities, it can be helpful in some cases.